### ISSUE TITLE
What's the point of standardization if barely anything is standardized?

### ISSUE THREAD

**Issue by procedural** [KhronosGroup/Vulkan-Docs#1259]
Reading about the OpenCL 3.0 fiasco made me think about the possible Vulkan future of no driver support and the current state of Vulkan.

If one actually programs in Vulkan, he/she may notice that almost every feature or a thing has a limit that is supposedly defined, but in practice [is a subject to change](https://github.com/KhronosGroup/Vulkan-Docs/issues/1044), and the features are all optional, and even though the standard increments like 1.1 require to implement some KHR extensions, those extensions may be ["implemented" with their feature flags set to 0](https://github.com/KhronosGroup/Vulkan-Docs/issues/1161#issuecomment-574298824).

So I thought: "well, if you don't really standardize anything other than how procedures, enums and structs look, what's the point of your standard then?". Joe, a Khronos member, will wake up the next morning and decide to deprecate Vulkan and define a new standard called Kraken by just renaming procedures, enums and structs, adding new synchronization primitives no one will conform to, implicitly blocking the thread like the current Vulkan WSI procedures do, and no one will stop him because Khronos people think this is what a standard is?

If your standard is full of variable limits, optional features and weak synchronization guarantees (no Vulkan release requires `VkPhysicalDeviceVulkanMemoryModelFeatures::vulkanMemoryModel` set to `1`), what *is* standardized other than how procedures look and what inputs they expect?

**Comment by procedural** [comment]
"Vulkan Portability attempts to make Vulkan available everywhere by layering it on top of Metal, D3D12, and others. Now, everybody starts using Vulkan, celebrate, and never look back, right? Not exactly. Vulkan may be fast, but its definition of “portable” is quite weak. Developing an app that doesn’t trigger Vulkan validation warnings(!) on any platform is extremely challenging. Reading Vulkan spec is exciting and eye opening in many respects, but applying it in practice is a different experience. Vulkan doesn’t provide low-level API to all GPU hardware, it has the head of a lion, the body of a bear, and the tail of a crocodile. Each piece of Vulkan matches some hardware (e.g. vkImageLayout matches AMD’s), but the other hardware treats it as a no-op at best, and as a burden at worst. It’s a jack of all trades.

Besides, real Vulkan isn’t everywhere. More specifically, there are no drivers for Intel Haswell/Broadwell iGPUs on Windows, it’s forbidden on Windows UWP (including ARM), it’s below 50% on Android, and totally absent on macOS and iOS. Vulkan Portability aims to solve it, but it’s another fairly complex layer for your application, especially considering the shader translation logic of SPIRV-Cross, and it’s still a WIP.

In gfx-rs community, we made a huge bet on Vulkan API when we decided to align our low-level Rusty API to it (see our Fosdem 2018 talk) instead of figuring out our own path. But when we (finally) understood that Vulkan API is simply unreachable by most users, we also realized that WebGPU on native is precisely the API we need to offer them."

http://kvark.github.io/web/gpu/native/2020/05/03/point-of-webgpu-native.html
